feat(tutorial): add step-2 (#1373)
carbon-design-system/carbon-for-ibm-dotcom-website: src/pages/developing/web-components-tutorial/step-2.mdx

```

<InlineNotification>
  **Note:** This builds on top of step 1, but be sure to check out the upstream
  step 2 branch because it includes the static assets required to get through
  this step.
</InlineNotification>

## Component documentation

In this step, you'll be adding 4 new components to the page; [Leadspace with search](https://www.ibm.com/standards/carbon/web-components/?path=/story/components-lead-space-search--default),
[Table of contents - horizontal](https://www.ibm.com/standards/carbon/web-components/?path=/story/components-table-of-contents--horizontal),
[Card section offset](https://www.ibm.com/standards/carbon/web-components/?path=/story/components-card-section-offset--default), and

> [CTA section with linked list](https://www.ibm.com/standards/carbon/web-components/?path=/story/components-cta-section--with-link-list). All components have

Reviewer: ```suggestion
[CTA section with link list](https://www.ibm.com/standards/carbon/web-components/?path=/story/components-cta-section--with-link-list). All components have
```

implementation details in the linked Storybook Docs tab. The Docs tab has a link to the codesandbox example of each component, props table to showcase different variations possible, and other
key information. This is a great place to look back to if you're ever stuck.

<InlineNotification>
  **Note:** We’ve included the designs for this tutorial app in the `design.pdf`
  file found as a top-level file in the carbon-for-ibm-dotcom-tutorial
  repository. You may notice there are additional components not listed above,
  but dont' worry those additional components we'll be adding in Step 3.
</InlineNotification>

## Add Leadspace with search

First let's incorporate `Leadspace with search` in our page. This can be done in two easy steps.

Import the following into `app.js`:

```javascript path=src/assets/js/app.js

> import '@carbon/ibmdotcom-web-components/es/components/dotcom-shell/index.js';

Reviewer: ```suggestion
import '@carbon/ibmdotcom-web-components/es/components/leadspace-with-search/index.js';
```

<!-- prettier-ignore-start -->
```html path=src/index.html
<div class="bx--grid">
  <div class="bx--row">
    <div class="bx--col-lg-8 bx--offset-lg-4 bx--no-gutter">
      <dds-leadspace-with-search>
        <dds-leadspace-block-heading slot="heading"
          >Find analytics services</dds-leadspace-block-heading
          >
        <dds-leadspace-block-content slot="content">
          <dds-leadspace-search-block-heading
            >Make fast and accurate data-based
            decisions
          </dds-leadspace-search-block-heading
            >
        </dds-leadspace-block-content>
        <dds-search-with-typeahead
          slot="search"
          leadspace-search></dds-search-with-typeahead>
      </dds-leadspace-with-search>
    </div>
  </div>
</div>
```
<!-- prettier-ignore-end -->

Nice! You have just added the `Leadspace with search` component to the page! But wait...the color of the component doesn't match the design.

> Looking at the props table in the documentation, there is a `adjacent-theme` property we can pass to the component to adjust the theme.

Reviewer: ```suggestion
Looking at the props table in the storybook documentation, there is an `adjacent-theme` property we can pass to the component to adjust the theme.
```

Let's do that by setting `adjacent-theme="g90-and-g100"` in `dds-leadspace-with-search`:

<!-- prettier-ignore-start -->
```html path=src/index.html
  <dds-leadspace-with-search adjacent-theme="g90-and-g100">
```
<!-- prettier-ignore-end -->

Now `Leadspace with search` should display the proper theme. Great job! Let's move on to the next component.

## Add Table of contents - horizontal

`Table of contents` is a great component to structure the page. We'll be using the horizontal variation in this step.

> Let's added the import to the `app.js` file:

Reviewer: ```suggestion
Let's add the import to the `app.js` file:
```

    <a name="1" data-title="First Item"></a>
    <h1>I am the first item in the table of contents</h1>
  </dds-table-of-contents>
```
<!-- prettier-ignore-end -->

You should see the `Table of contents` with one item listed. Each item in the `Table of contents` will need a preceding anchor element
with a unique `name` and title of the item as`data-title.` That's it!

## Add Card section offset

Now within the `Table of contents`, we'll set up the `Card section offset` component. This component requires image assets that are provided
in the `src/assets/images` folder.

Import the following in `app.js`:

```javascript path=src/assets/js/app.js
import '@carbon/ibmdotcom-web-components/es/components/card-section-offset/index.js';
```

Then replace the following within the `Table of contents`:

<!-- prettier-ignore-start -->
```html path=src/index.html
    <a name="1" data-title="First Item"></a>
    <h1>I am the first item in the table of contents</h1>
```
<!-- prettier-ignore-end -->

With the `Card section offset`, noting the anchor element precedding that is included for the `Table of contents`:

<!-- prettier-ignore-start -->
```html path=src/index.html

> <a name="1" data-title="Main benefits of analytics"></a>

Reviewer: Maybe we can add something like. Since this is the 3rd anchor in the design we can assign name="3"
```suggestion
<a name="3" data-title="Main benefits of analytics"></a>
```

such as `Background media`, `Card`, `Card Group`, etc which have their own set of properties that can be tweaked. Check out the
Storybook Docs for them for more information!

## Add CTA section with link list

With the `CTA section` component in particular, we'll be setting up theming that can be applied to a certain component,
entire sections, or a page. To incorporate the g100 theme to our component, we'll create a class in our `app.scss` file and wrap it around our `CTA section`. For more details you can go to Carbon's
[theme overview](https://www.carbondesignsystem.com/guidelines/themes/overview/) page. This class, `.dark-theme` can then be used to wrap several components or sections of the page.

```css path=src/assets/scss/app.scss
@import '@carbon/themes/scss/themes';

.dark-theme {
  @include carbon--theme($carbon--theme--g100, true);

  background-color: var(--cds-ui-background);
  color: var(--cds-text-01);
}
```

Then create a `<div>` element to wrap our `CTA section` component. This component is also going to be within the `Table of contents` so remember
to add the preceding anchor element:

<!-- prettier-ignore-start -->
```html path=src/index.html

> <a name="2" data-title="Engage an expert"></a>

Reviewer: same suggestion as before
```suggestion
<a name="5" data-title="Engage an expert"></a>
```
